pulp_python/tests/functional/api/test_download_content.py

```diff
@@ -87,3 +87,30 @@ def test_full_pulp_to_pulp_sync(
     repo3 = python_repo_with_sync(remote3)
     summary2 = python_content_summary(repository_version=repo3.latest_version_href)
     assert summary2.present["python.python"]["count"] == PYTHON_MD_PACKAGE_COUNT
+
+
+@pytest.mark.parallel
+def test_pulp2pulp_sync_with_oddities(
+    python_repo_with_sync,
+    python_remote_factory,
+    python_publication_factory,
+    python_distribution_factory,
+    python_content_summary,
+):
+    """Test that Pulp can handle syncing packages with wierd names."""
+    remote = python_remote_factory(includes=["oslo.utils"], url="https://pypi.org")
+    repo = python_repo_with_sync(remote)
+    distro = python_distribution_factory(repository=repo)
+    # Test pulp 2 pulp full sync w/ live pypi apis
+    remote2 = python_remote_factory(includes=[], url=distro.base_url)
+    repo2 = python_repo_with_sync(remote2)
+    summary = python_content_summary(repository_version=repo2.latest_version_href)
+    assert summary.present["python.python"]["count"] > 0
```

Comment:
I'd love to see a bit more explanation here, why this assertion actually tells me that everything is ok. Maybe we need a short explanation of the bug that was fixed to describe that accurately.
Should we not also see the exact same number of content as in the `repo` here?

Reply:
You're right we should see the same amount, forgot to add the check.